blockly-core/core/block.js

```diff
@@ -1219,6 +1219,22 @@ Blockly.Block.prototype.getDescendants = function() {
 };
 
 /**
+ * Get whether this block and all of its connected descendants
+ * are configured as deletable. Used when deciding whether the
+ * user can trash a group of blocks - all of the blocks must
+ * be deletable.
+ * @return {boolean} True if this block and all blocks connected
+ *    below it are deleteable.
+ */
+Blockly.Block.prototype.areBlockAndDescendantsDeletable = function() {
+  var deleteBlockedByChildren = this.childBlocks_.some(
+    function (child) {
+      return !child.areBlockAndDescendantsDeletable();
+    });
+  return !deleteBlockedByChildren && this.isDeletable();
```

Comment:
complete nit, but my brain finds this a lot easier to parse if we flip the order (this.isDeletable() && !deleteBlockedByCHildren)